app/controllers/user_sessions_controller.rb

```diff
@@ -26,8 +26,14 @@ def handle_social_login_flow(auth)
     unless params[:hash_params].to_s.empty?
       hash_params = URI.parse("#" + params[:hash_params]).to_s
     end
-
-    if signed_in?
+    @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : @identity.user
+    if @user&.status&.zero?
+      flash[:error] = I18n.t('user_sessions_controller.user_has_been_banned', username: @user.username).html_safe
+      redirect_to return_to + hash_params
```

Comment:
We need a test to ensure this redirection too.

Reply:
Yeah but the problem here is that as mentioned in #6233, oauth is not redirecting correctly. I will try to understand why

Reply:
This is the test I have for now but it won't pass because none of the redirection for oauth is working:
```ruby
  test "if banned user logs in through omniauth, should redirect properly" do
    request.env['omniauth.origin'] = "/notes/liked"
    request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:github1]
    # create account
    post :create
    # logout
    post :destroy
    User.find_by(name: "bansal_sidharth309").ban
    post :create
    assert_redirected_to "/notes/liked"
  end
```

Reply:
We can handle oauth redirection in other pr. That PR should be merged before this pr